python/flip/main.cpp

```diff
@@ -0,0 +1,284 @@
+void updateInputParameters(const FLIP::Parameters& parameters, py::dict& inputParameters)
+{
+    inputParameters["ppd"] = parameters.PPD;
+    inputParameters["startExposure"] = parameters.startExposure;
+    inputParameters["stopExposure"] = parameters.stopExposure;
+    inputParameters["numExposures"] = parameters.numExposures;
+    inputParameters["tonemapper"] = parameters.tonemapper;
+}
+
+/** A simplified function for evaluating (the image metric called) FLIP between a reference image and a test image. Corresponds to the fourth evaluate() option in FLIP.h.
+*
+* @param[in] referenceInput Reference input image. For LDR, the content should be in [0,1]. The image is expected to have 3 floats per pixel.
+* @param[in] testInput Test input image. For LDR, the content should be in [0,1].
+* @param[in,out] parameters Contains parameters (e.g., PPD, exposure settings,etc). If the exposures have not been set by the user, then those will be computed (and returned).
+* @param[in] useHDR Set to true if the input images are to be considered containing HDR content, i.e., not necessarily in [0,1].
+* @param[in] applyMagmaMapToOutput A boolean indicating whether the output should have the MagmaMap applied to it before the image is returned.
+* @param[out] errorMapFLIPOutput The computed FLIP error image is returned in this variable. If applyMagmaMapToOutput is true, the function will allocate
+*             three channels (and store the magma-mapped FLIP images in sRGB), and
+*             if it is false, only one channel will be allocated (and the FLIP error is returned in that grayscale image).
+*             Note that the user is responsible for deallocating the errorMapFLIPOutput image.
+*/
+std::tuple<py::array_t<float>, float, py::dict> evaluate(const py::array_t<float> referenceInput, const py::array_t<float> testInput, const bool inputHDR, const bool inputsRGB = true, const bool returnMeanError = true, const bool applyMagma = true, py::dict inputParameters = {})
+{
+    py::buffer_info r_buf = referenceInput.request(), t_buf = testInput.request();
+    
+    // Check number of dimensions and resolution.
+    if (r_buf.ndim != 3 || t_buf.ndim != 3)
+        throw std::runtime_error("Number of dimensions must be three.");
+    if (r_buf.shape[0] != t_buf.shape[0] || r_buf.shape[1] != t_buf.shape[1] || r_buf.shape[2] != t_buf.shape[2])
+        throw std::runtime_error("Reference and Test image resolutions differ.");
+    
+    // Arrays for reference and test.
+    float* ptr_r = static_cast<float*>(r_buf.ptr);
+    float* ptr_t = static_cast<float*>(t_buf.ptr);
+
+    // Image size.
+    const int nRows = int(r_buf.shape[0]), nCols = int(r_buf.shape[1]), nChannels = int(r_buf.shape[2]);
+
+    // FLIP
+    float* flip;
+
+    // Create NumPy output array.
+    py::array_t<float> flipNumpy;
+    int nChannelsOut;
+    if (applyMagma)
+    {
+        flipNumpy = py::array_t<float>({ r_buf.shape[0], r_buf.shape[1], r_buf.shape[2] });
```

Comment:
Shouldn't the last arg be 3 instead of ` r_buf.shape[2]`? Probably gives the same result, but  feels easier to read with 3.

Reply:
The `py::array_t<float>` constructor does not take integers in this case, so I'll keep the `r_buf.shape[2]`.